subprojects/diagnostics/src/main/java/org/gradle/configuration/Help.java

```diff
@@ -36,18 +38,17 @@
 
 @DisableCachingByDefault(because = "Produces only non-cacheable console output")
 public abstract class Help extends DefaultTask {
-    private final Property<String> taskPath = getObjectFactory().property(String.class);
-
-    private final Property<TaskDetailsModel> taskModel = getObjectFactory().property(TaskDetailsModel.class).convention(taskPath.map(this::mapFromTaskPath));
+    private final Property<TaskDetailsModel> taskModel = getObjectFactory().property(TaskDetailsModel.class).convention(getTaskPath().map(this::mapFromTaskPath));
 
     public Help() {
         // optimization: so value does not need to be recomputed during execution
         taskModel.finalizeValueOnRead();
     }
 
-    private Property<String> getTaskPath() {
-        return taskPath;
-    }
+    @Input
+    @Optional
+    @Option(option = "task", description = "The task to show help for.")
+    public abstract Property<String> getTaskPath();
```

Comment:
❓  Hmm, shouldn't we use `@ReplacesEagerProperty` to upgrade a setter

Reply:
Well, it was only public because we needed it to set it via CLI. 
Is it valuable to preserve the setter at all?

Reply:
I personally would upgrade it, since cost is close to zero. And if it's there, I guess somebody is using it.

I wonder if there are users/plugins that do something like:
```
tasks.register<Help>("myTaskHelp") {
     setTaskPath("myTask")
}
```

Reply:
Hmm, interesting.
With
```
    @ReplacesEagerProperty(replacedAccessors = {@ReplacedAccessor(value = ReplacedAccessor.AccessorType.SETTER, name = "setTaskPath")})

```
I get

```
 The following accessors were upgraded, but didn't match any removed/changed method:
     
     org.gradle.configuration.Help#setTaskPath(Ljava/lang/String;)V
```

Reply:
What 🤔

Reply:
Actually it seems Help is not public API, so that is not a problem as initially implemented, my bad. Resolving thi